packages/client/hmi-client/src/components/workflow/ops/model-from-equations/tera-model-from-equations-drilldown.vue

```diff
@@ -379,20 +406,82 @@ function updateNodeLabel(id: string, label: string) {
 	emit('update-output-port', outputPort);
 }
 
-function addEquation() {
-	clonedState.value.equations.push({
-		name: 'Equation',
-		includeInProcess: true,
-		asset: {
-			text: ''
+function removeEquation(index: number) {
+	clonedState.value.equations.splice(index, 1);
+	emit('update-state', clonedState.value);
+}
+
+const menuItems = computed(() => [
+	{
+		label: 'Save as new model',
+		icon: 'pi pi-download',
+		disabled: !selectedModel.value,
+		command: () => {
+			showSaveModelModal.value = true;
 		}
+	}
+]);
+
+function getEquations() {
+	const newEquations = multipleEquations.value.split('\n');
+	newEquations.forEach((equation) => {
+		clonedState.value.equations.push({
+			name: 'Equation',
+			includeInProcess: true,
+			asset: {
+				text: equation
+			}
+		});
 	});
 	emit('update-state', clonedState.value);
+	multipleEquations.value = '';
 }
 
-function removeEquation(index: number) {
-	clonedState.value.equations.splice(index, 1);
-	emit('update-state', clonedState.value);
+const allEquationCollapsed = computed(
+	() => !clonedState.value.equations.some((equation) => !equation.collapsed)
+);
+
+const allEquationsInProcess = computed(
+	() => !clonedState.value.equations.some((equation) => !equation.includeInProcess)
+);
+
+const selectedEquations = computed(() =>
+	clonedState.value.equations.filter((equation) => equation.includeInProcess)
+);
+
+function getEquationErrorLabel(equation) {
+	return equation.asset.extractionError ? "Couldn't extract equation" : '';
+}
+
+function getEquationSelectedLabel() {
+	const total = clonedState.value.equations.length;
+	return `(${selectedEquations.value.length}/${total} selected)`;
+}
+
+function getCollapsedLabel() {
+	return allEquationCollapsed.value ? 'Expand All' : 'Collapse all';
+}
+
+function getIncludedEquationLabel() {
+	return allEquationsInProcess.value ? 'Remove all from process' : 'Include all in process';
+}
+
+function changeCollapsed(equation, isCollapsed = false) {
+	equation.collapsed = isCollapsed;
+}
+
+function toggleCollapseAll() {
+	const collapseEquations = allEquationCollapsed.value;
+	clonedState.value.equations.forEach((equation) => {
+		changeCollapsed(equation, !collapseEquations);
+	});
+}
+
+function toggleIncludedEquations() {
+	const allEquationsIncluded = allEquationsInProcess.value;
+	clonedState.value.equations.forEach((equation) => {
+		equation.includeInProcess = !allEquationsIncluded;
+	});
 }
```

Comment:
Is there a need to copy `allEquationCollapsed.value` and `allEquationsInProcess.value` instead of just using them directly?

Reply:
Since the `allEquationCollapsed`  and `allEquationsInProcess` are computed based off of the list of equations. That value will change once the first value in the list is changed. This would cause the second `boolean` being assigned to be wrong.